plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -154,6 +154,7 @@ class chain_plugin_impl {
    fc::microseconds                  abi_serializer_max_time_us;
    std::optional<bfs::path>          snapshot_path;
 
+   std::shared_ptr<chain_plugin_metrics> chain_metrics;
```

Comment:
What necessitates this being a `shared_ptr` over a `unique_ptr`, or even just returning a reference to a `const chain_plugin_metrics&`?